### ISSUE TITLE
[Regression] jsdoc/valid-types for |undefined properties

### ISSUE THREAD

**Issue by kevinoid** [gajus/eslint-plugin-jsdoc#749]
## Expected behavior

`jsdoc/valid-types` does not warn for `{message: string|undefined}`.

## Actual behavior

    index.js:2:0: Syntax error in type: {message: string|undefined} [Warning/jsdoc/valid-types]

## ESLint Config

```json
{
  "extends": [
    "plugin:jsdoc/recommended"
  ]
}
```

## ESLint sample

```js
/**
 * @param {{message: string|undefined}} options Options.
 */
function MyError(options) {
}
```

## Environment

- Node version: v12.21.0
- ESLint version v7.27.0
- `eslint-plugin-jsdoc` version: v35.1.1 or v35.1.0

## Additional Info

I confirmed the issue does not occur with `eslint-plugin-jsdoc@35.0.0` or earlier.

Thanks again,
Kevin

**Comment by brettz9** [comment]
Thanks for all the reports!

**Comment by kevinoid** [comment]
My pleasure.  Thanks for all of your work fixing them (and maintaining the project)!
